File: frontend/src/types/budget.ts
export interface Budget {
  id: number
  category: string
  monthlyLimit: number
}

export interface BudgetProgressItem {
  category: string
  monthlyLimit: number
  spent: number
  remaining: number
  percentUsed: number
}

export interface BudgetProgressResponse {
  year: number
  month: number
  items: BudgetProgressItem[]
}

File: backend/src/main/java/com/finance/personel_finance/budget/model/entity/Budget.java
package com.finance.personel_finance.budget.model.entity;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(
        name = "budgets",
        uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "year", "month", "category"})
)
public class Budget {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(nullable = false)
    private int year;

    @Column(nullable = false)
    private int month; // 1-12

    @Column(nullable = false, length = 50)
    private String category;

    @Column(name = "monthly_limit", nullable = false, precision = 12, scale = 2)
    private BigDecimal monthlyLimit;
}